DEVELOP.md

```diff
@@ -0,0 +1,139 @@
+# Development
+
+This document will guide you through how to
+. Set up the local development environment
+. Build web UI and configure backend service
+. Run the cloud gaming demo for the local development
+. Add additional games
+
+## Web UI
+
+The web UI for the clouding gaming demo is built with [Flutter](https://flutter.dev/).
+To set up the development environment and build the UI, make sure that you have Flutter installed on your machine.
+Follow the [installation instructions](https://docs.flutter.dev/get-started/install/linux) to install it.
+
+### How to build
+
+To build the UI, simply run:
+
+  ```
+  cd ui && flutter build web
+  ```
+This builds the UI and places outputs in the `ui/build/web` directory.
+Refer to the official [documentation](https://docs.flutter.dev/get-started/web) for details on how to build web apps with Flutter.
+
+### How to deploy
+
+Create a `static` folder under the `backend/service` directory and copy everything that is in the `ui/build/web` directory into it.
+
+  ```
+  mkdir -p backend/service/static
+  cp -av ui/build/web/* backend/service/static
+  ```
+Those static resources are then served by backend service when you access the web UI.
+
+
+## Backend
+
+The backend service is not just for serving the web UI, but also interacts with Anbox Stream Gateway and provides a set of restful APIs for clients(web UI) to perform operations like list games or launch a game.
+
+### Install the dependency
+
+Install the `virtualenv` package to create an isolated python environment that the backend service can run on
+
+  ```
+  sudo apt-get install python3-pip
+  sudo pip3 install virtualenv
+  cd backend && virtualenv venv
+  ```
+
+### Configure the service
+
+To access the APIs that are exposed by Anbox Stream Gateway, the backend service needs to know
+
+. The URL of Anbox Stream Gateway
+. The access token for Anbox Stream Gateway Restful API authentication
+
+As Anbox Stream Gateway runs behind traefik and secures the Restful API with HTTPS, its URL can be composed by;
+
+  ```
+  eval "$(anbox-cloud-appliance internal generate-cloud-info)"
+  gateway_url="https://${CLOUD_PUBLIC_LOCATION}"
+  ```
+Create an access token to authenticate API requests from a client:
+
+   ```
+   gateway_token=$(sudo -u ubuntu juju ssh -m appliance:anbox-cloud anbox-stream-gateway/0 \
+                   -- "sudo /snap/bin/anbox-stream-gateway account create <account-name>")
+   ```
+Then create a file called `config.yaml` that contains those two config items
+
+   ```
+   gateway-url: <gateway_url>
+   gateway-token: <gateway_token>
+   ```
+Add the following line to the `activate` hook under `backend/venv/bin/` directory so that the backend service can read the `config.yaml` on its startup.
```

Comment:
```suggestion
Add the following line to the `activate` hook in the `backend/venv/bin/` directory so that the backend service can read the `config.yaml` on startup:
```